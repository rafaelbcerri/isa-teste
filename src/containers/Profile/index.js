import React, { useEffect, useState } from 'react';
import axios from 'axios';

import Card from '../../componentes/Card';
import { formatCurrency } from '../../util';

function Profile() {
  const [ user, setUser ] = useState();

  useEffect(() => {
    axios
      .get('https://jsonplaceholder.typicode.com/users/1')
      .then(({ data }) => setUser(data))
  }, [])

  const handleClick = () => {
    alert('Hello')
  }

  if (!user) {
    return <h1>Loading...</h1>
  }

  return (
    <>
      <h1>Hello</h1>
      <Card
        title={user.name}
        text={user.email}
        onClick={handleClick}
      />
      <p>Banana price: {formatCurrency(2.50)}</p>
    </>
  )
}

export default Profile;