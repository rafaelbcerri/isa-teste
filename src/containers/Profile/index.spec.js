import axios from 'axios';
import {
  render,
  waitFor,
  cleanup,
  screen,
  fireEvent,
  act
} from '@testing-library/react';

import Profile from '.';

describe.only('Profile', () => {
  afterEach(() => {
    jest.clearAllMocks();
    cleanup()
  })

  describe('when component starts', () => {
    it('should show Loading text', () => {
      let container;

      act(() => {
        const rootElem = render(<Profile />);
        container = rootElem.container;
      });

      expect(container).toMatchSnapshot();
    });
  })

  describe('when user is loaded', () => {
    it('should render', async () => {
      let container;
      const axiosGetSpy = jest
        .spyOn(axios, 'get')
        .mockResolvedValueOnce({ data: {
          name: 'Robot',
          email: 'robot@example.com'
        }});

      act(() => {
        const rootElem = render(<Profile />);
        container = rootElem.container;
      });

      await waitFor(() => expect(axiosGetSpy).toBeCalled());

      expect(container).toMatchSnapshot();
    });

    describe('and card button is clicked', () => {
      it('should open an alert', async () => {
        window.alert = jest.fn();
        axios.get = jest.fn().mockResolvedValueOnce({
          data: { name: 'Robot', email: 'robot@example.com' }
        });

        act(() => {render(<Profile />)});
        await waitFor(() => expect(axios.get).toBeCalled());
        act(() => {fireEvent.click(screen.getByText(/Enviar/))});

        expect(window.alert).toBeCalled();
      });
    });
  });
});