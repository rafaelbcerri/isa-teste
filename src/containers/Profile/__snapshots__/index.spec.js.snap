// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile when component starts should show Loading text 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;

exports[`Profile when user is loaded should render 1`] = `
<div>
  <h1>
    Hello
  </h1>
  <div
    class="card"
  >
    <h1>
      Robot
    </h1>
    <hr />
    <p>
      robot@example.com
    </p>
    <button>
      Enviar
    </button>
    <button>
      Esconde titulo
    </button>
  </div>
  <p>
    Banana price: 
    R$ 2.5
  </p>
</div>
`;
